[sparsity] Add axis support to the pruning parametrization

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66416
* __->__ #66415
* #66414
* #66413
* #66412
* #66411

We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

Differential Revision: [D31590677](https://our.internmc.facebook.com/intern/diff/D31590677)

Commit:
[sparsity] Add axis support to the pruning parametrization

We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

[ghstack-poisoned]

Commit:
Update on "[sparsity] Add axis support to the pruning parametrization"


We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

Differential Revision: [D31590677](https://our.internmc.facebook.com/intern/diff/D31590677)

[ghstack-poisoned]

HDCharles reviewed (commented):
can we use https://pytorch.org/docs/stable/generated/torch.Tensor.select.html#torch.Tensor.select to get around hard coding the axis to 1 or 0?

Commit:
Update on "[sparsity] Add axis support to the pruning parametrization"


We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

Differential Revision: [D31590677](https://our.internmc.facebook.com/intern/diff/D31590677)

[ghstack-poisoned]

Commit:
Update on "[sparsity] Add axis support to the pruning parametrization"


We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

Differential Revision: [D31590677](https://our.internmc.facebook.com/intern/diff/D31590677)

[ghstack-poisoned]

Commit:
Update on "[sparsity] Add axis support to the pruning parametrization"


We would like to allow the users to use any axis for pruning.
THis allows the base pruner to attach axis config to the pruners.

Differential Revision: [D31590677](https://our.internmc.facebook.com/intern/diff/D31590677)

[ghstack-poisoned]

z-a-f commented:
@zafartahirov has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31590677).
